### PR TITLE
STYLE,BUG: autotyping hook fails if run on no files

### LINKED ISSUE #48808: STYLE,BUG: autotyping hook fails if run on no files
Not sure why yet, but here's what I got after a commit:
```
(pandas-dev) marcogorelli@DESKTOP-U8OKFP3:~/pandas-dev/doc$ git commit -a -m 'doc'
absolufy-imports........................................................................................Passed
vulture.................................................................................................Passed
black...................................................................................................Passed
codespell...............................................................................................Passed
cython-lint.........................................................................(no files to check)Skipped
debug statements (python)...............................................................................Passed
fix end of files........................................................................................Passed
trim trailing whitespace................................................................................Passed
cpplint.............................................................................(no files to check)Skipped
flake8..................................................................................................Passed
isort...................................................................................................Passed
pyupgrade...............................................................................................Passed
rst ``code`` is two backticks.......................................................(no files to check)Skipped
rst directives end with two colons......................................................................Passed
rst ``inline code`` next to normal text.................................................................Passed
Sphinx lint.........................................................................(no files to check)Skipped
Strip unnecessary `# noqa`s.............................................................................Passed
flake8-rst..........................................................................(no files to check)Skipped
Unwanted patterns.......................................................................................Passed
Check Cython casting is `<type>obj`, not `<type> obj`...............................(no files to check)Skipped
Check for backticks incorrectly rendering because of missing spaces.................(no files to check)Skipped
Check for unnecessary random seeds in asv benchmarks................................(no files to check)Skipped
Check for usage of numpy testing or array_equal.....................................(no files to check)Skipped
Check for invalid EA testing........................................................(no files to check)Skipped
Generate pip dependency from conda..................................................(no files to check)Skipped
Check flake8 version is synced across flake8, yesqa, and environment.yml............(no files to check)Skipped
Validate correct capitalization among titles in documentation.......................(no files to check)Skipped
Import pandas.array as pd_array in core.................................................................Passed
Use pandas.io.common.urlopen instead of urllib.request.urlopen..........................................Passed
Use bool_t instead of bool in pandas/core/generic.py................................(no files to check)Skipped
Ensure pandas errors are documented in doc/source/reference/testing.rst.............(no files to check)Skipped
Check for pg8000 not installed on CI for test_pg8000_sqlalchemy_passthrough_error...(no files to check)Skipped
Check minimum version of dependencies are aligned...................................(no files to check)Skipped
Validate errors locations...............................................................................Passed
flake8-pyi..........................................................................(no files to check)Skipped
import annotations from __future__......................................................................Passed
autotyping..............................................................................................Failed
- hook id: autotyping
- exit code: 2


usage: libcst.tool codemod
       [-h]
       [-j JOBS]
       [-p VERSION]
       [-u [CONTEXT]]
       [--include-generated]
       [--include-stubs]
       [--no-format]
       [--show-successes]
       [--hide-generated-warnings]
       [--hide-blacklisted-warnings]
       [--hide-progress]
       [--annotate-optional [ANNOTATE_OPTIONAL [ANNOTATE_OPTIONAL ...]]]
       [--annotate-named-param [ANNOTATE_NAMED_PARAM [ANNOTATE_NAMED_PARAM ...]]]
       [--none-return]
       [--scalar-return]
       [--bool-param]
       [--int-param]
       [--float-param]
       [--str-param]
       [--bytes-param]
       [--annotate-magics]
       [--annotate-imprecise-magics]
       [--pyanalyze-report PYANALYZE_REPORT]
       [--only-without-imports]
       [--safe [SAFE]]
       [--aggressive [AGGRESSIVE]]
       COMMAND
       PATH
       [PATH ...]
libcst.tool codemod: error: the following arguments are required: PATH
```

This hook was added in https://github.com/pandas-dev/pandas/pull/48191
cc @twoertwein in case you've come across this before (I'll have a look later anyway, just opening an issue for now)

### PR DESCRIPTION
**Pull request by MarcoGorelli**
- [ ] closes #48808 (Replace xxxx with the Github issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: scripts/run_autotyping.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+            "codemod",
+            "autotyping.AutotypeCommand",
+            *args.paths,
+            "--aggressive",
+        ],
+        capture_output=True,
+        text=True,
+    )
+    sys.stdout.write(output.stdout)
+    sys.stdout.write(output.stderr)
```
Comment (twoertwein) [inline scripts/run_autotyping.py:33]: Has this the same effect as removing `capture_output=True` and removing those two write statements?

Comment (MarcoGorelli) [reply scripts/run_autotyping.py]: yup - turns out I'd always been overcomplicating using `subprocess.run`, thanks!

File: scripts/run_autotyping.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import subprocess
+import sys
+from typing import Sequence
+
+
+def main(argv: Sequence[str] | None = None) -> None:
+    parser = argparse.ArgumentParser()
+    parser.add_argument("paths", nargs="*")
+    args = parser.parse_args(argv)
+    output = subprocess.run(
```
Comment (twoertwein) [inline scripts/run_autotyping.py:19]: Should proably add 

```py
if not args.paths:
    sys.exit(0)
```

When I run `python -m scripts.run_autotyping` I still get the error about missing paths.

**Review by twoertwein — APPROVED** [review]

**Comment by twoertwein** [comment]
Thanks @MarcoGorelli !

**Comment by MarcoGorelli** [comment]
should probably backport to avoid break people's contribution flow in 1.5.x

@meeseeksdev backport 1.5.x

**Comment by MarcoGorelli** [comment]
@meeseeksdev backport 1.5.x

**Comment by MarcoGorelli** [comment]
nevermind, 1.5.x never had autotyping to begin with, no need to backport

**Pull request merged**